core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/controller/promisehandlers/SkewDetectionHandler.scala

```diff
@@ -241,7 +241,7 @@ object SkewDetectionHandler {
     if (workflow.physicalPlan.getOperator(physicalOpId).isHashJoinOperator) {
       upstreamPhysicalOps
         .find(physicalOp => {
-          val buildTableLinkId = physicalOp.inputPortToLinkMapping(0).head.id
+          val buildTableLinkId = physicalOp.inputPortToLinkMapping(0).head
```

Comment:
Is this one a link or an ID? If it is a link, better rename the variable.

Reply:
it is a linkId. I have renamed the mapping name.